src/components/KPop/KPop.vue

```diff
@@ -1,431 +1,306 @@
+}
 
-      await this.$nextTick()
-      this.popper = new Popper(this.reference, popperEl, {
-        placement,
-        // Use positionFixed to avoid popover content being cut off by parent boundaries
-        positionFixed: this.positionFixed,
-        modifiers: {
-          // Ensures element does not ovflow outside of boundary
-          preventOverflow: {
-            enabled: true,
-            boundariesElement: 'viewport',
-          },
-        },
-      })
-      await this.$nextTick()
-      if (theTarget) {
-        // Remove overflow attribute now that rendering is complete
-        theTarget.style.removeProperty('overflow')
-      }
-      this.updatePopper()
-    },
-    handleClick(e) {
-      const hidePopperAndStopPropagation = () => {
-        // Stop event propagation only when the click event is about to hide popper
-        e.stopPropagation()
-        this.hidePopper()
-      }
+const onPopoverClick = () => {
+  if (props.closeOnPopoverClick) {
+    hidePopover()
+  }
 
-      if (e.target === this.$refs.popoverCloseButton) {
-        hidePopperAndStopPropagation()
-      }
+  emit('popover-click')
+}
 
-      if (this.reference && this.reference.contains(e.target) && (this.$refs.popper && !this.$refs.popper.contains(e.target))) {
-        // If the click is on or within the trigger element and not on the popover element
-        if (this.isOpen) {
-          hidePopperAndStopPropagation()
-        } else {
-          this.createInstance()
-        }
-      } else if (this.$refs.popper && this.$refs.popper.contains(e.target)) {
-        // If the click is on or within the popover element
-        if (this.closeOnPopoverClick) {
-          hidePopperAndStopPropagation()
-        }
-      } else if (this.isOpen) {
-        // Click outside
-        hidePopperAndStopPropagation()
+const clickHandler = (event: Event) => {
+  const target = event.target as HTMLElement
+
+  if (!kPopoverElement.value?.contains(target)) {
+    hidePopover()
+  }
+}
+
+const popoverStyle = computed(() => {
+  return {
+    width: getSizeFromString(props.width),
+    maxWidth: getSizeFromString(props.maxWidth),
+    maxHeight: getSizeFromString(props.maxHeight),
+    ...floatingStyles.value,
+  }
+})
+
+const popoverClassesObj = computed(() => [props.popoverClasses, { 'hide-caret': props.hideCaret }])
+
+const { floatingStyles, placement: calculatedPlacement } = useFloating(popoverTrigger, popoverElement, {
+  ...(props.placement === 'auto' ? { middleware: [autoPlacement()] } : { placement: props.placement }),
+  strategy: props.positionFixed ? 'fixed' : 'absolute',
+  /**
+   * floating-ui docs don't recommend using whileElementsMounted with v-show
+   * but otherwise the popover won't update its position even when using manual update method
+   * so we keep using autoUpdate for now
+   * Docs: https://floating-ui.com/docs/vue#anchoring
+   */
+  whileElementsMounted: autoUpdate,
+})
+
+onMounted(() => {
+  if (document) {
```

Comment:
You shouldn't need the document check here, onMounted is only called in the client. Doesn't hurt to leave it I guess